server/src/main/java/io/druid/guice/DruidProcessingModule.java

```diff
@@ -98,6 +100,21 @@ public ExecutorService getProcessingExecutorService(
   }
 
   @Provides
+  @QueryResourceWaitingPool
+  @LazySingleton
+  public ListeningExecutorService getQueryResourceWaitingPool() throws Exception
+  {
+    return MoreExecutors.listeningDecorator(
```

Comment:
if i am understanding it correct, its listeningexecutor so that the futures can be registered with QueryWatcher, 
see QueryManager.registerQuery()